src/java.base/unix/classes/sun/nio/fs/UnixPath.java

```diff
@@ -857,13 +860,83 @@ public Path toRealPath(LinkOption... options) throws IOException {
             result = result.resolve(element);
         }
 
-        // check file exists (without following links)
+        // check whether file exists (without following links)
         try {
             UnixFileAttributes.get(result, false);
         } catch (UnixException x) {
             x.rethrowAsIOException(result);
         }
-        return result;
+
+        // Return if the file system is not both case insensitive and retentive
+        if (!fs.isCaseInsensitiveAndRetentive())
```

Comment:
I think we need to find a better name for this method. Maybe "Preserving" rather than "Retentive" would work better here.

Reply:
Renamed in 49d18c6a39d9d70b7da91c5b4b3740ba3c890948.